xdsl/dialects/arm/register.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+
+from xdsl.backend.register_type import RegisterType
+from xdsl.ir import Attribute
+from xdsl.irdl import irdl_attr_definition
+from xdsl.parser import AttrParser
+from xdsl.utils.exceptions import VerifyException
+
+
+class ARMRegisterType(RegisterType):
+    """
+    The abstract class for all ARM register types.
+    """
+
+    @classmethod
+    def parse_parameters(cls, parser: AttrParser) -> Sequence[Attribute]:
```

Comment:
```suggestion
class ARMRegisterType(RegisterType, abc.ABC):
    """
    The abstract class for all ARM register types.
    """

class ARMIntRegisterType(ARMRegisterType):
    """
    ARM Integer register class.
    """
    
    name = "reg"
    
    @classmethod
    def parse_parameters(cls, parser: AttrParser) -> Sequence[Attribute]:
```

Reply:
We do want to group all ARM registers together, so it's useful to have an abstract base class. This split would make the `ARMRegisterType` actually abstract, and also provide an int register class. I have a feeling that we'll need to do something a bit more involved when we start supporting both X and W registers, but I think it should be OK to tackle that later.